uuvir corn orders.app/;   p?:app     :app --reload[?1l>[?2004l
]2;uvicorn orders.app:app --reload]1;uvicorn[32mINFO[0m:     Will watch for changes in these directories: ['/Users/corikomiyama/Desktop/coffee-mesh-project/ch02']
[32mINFO[0m:     Uvicorn running on [1mhttp://127.0.0.1:8000[0m (Press CTRL+C to quit)
[32mINFO[0m:     Started reloader process [[36m[1m5214[0m] using [36m[1mStatReload[0m
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 31
    def create_order(order_details: CreateOrderSchema):
IndentationError: unexpected indent
[33mWARNING[0m:  StatReload detected changes in 'orders/api/api.py'. Reloading...
Process SpawnProcess-2:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 19, in <module>
    class OrderItemSchema(BaseModel):   # Every pydantic model inherits from pydantic's BaseModel
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 22, in OrderItemSchema
    quantity: Optional[conint(ge=1, strict=True)] = 1   # We specify quantity's minimum value and we give it a default
              ^^^^^^^^
NameError: name 'Optional' is not defined
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-3:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 19, in <module>
    class OrderItemSchema(BaseModel):   # Every pydantic model inherits from pydantic's BaseModel
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 22, in OrderItemSchema
    quantity: optional[conint(ge=1, strict=True)] = 1   # We specify quantity's minimum value and we give it a default
              ^^^^^^^^
NameError: name 'optional' is not defined
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-4:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 19, in <module>
    class OrderItemSchema(BaseModel):   # Every pydantic model inherits from pydantic's BaseModel
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 22, in OrderItemSchema
    quantity: Optional[conint(ge=1, strict=True)] = 1   # We specify quantity's minimum value and we give it a default
              ^^^^^^^^
NameError: name 'Optional' is not defined
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-5:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 24, in <module>
    class CreateOrderSchema(BaseModel):
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 25, in CreateOrderSchema
    order: conlist(OrderItemSchema, min_items=1)    # We use pydantic's conlist type to define a list with at least one element
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: conlist() got an unexpected keyword argument 'min_items'
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-6:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 24, in <module>
    class CreateOrderSchema(BaseModel):
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 25, in CreateOrderSchema
    order: conlist(OrderItemSchema, min_items=1)    # We use pydantic's conlist type to define a list with at least one element
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: conlist() got an unexpected keyword argument 'min_items'
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-7:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 19, in <module>
    class OrderItemSchema(BaseModel):   # Every pydantic model inherits from pydantic's BaseModel
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 22, in OrderItemSchema
    quantity: Optional[conint(ge=1, strict=True)] = 1   # We specify quantity's minimum value and we give it a default
              ^^^^^^^^
NameError: name 'Optional' is not defined
clear
^C[32mINFO[0m:     Stopping reloader process [[36m[1m5214[0m]
[1m[7m%[27m[1m[0m                                                                                                                                                            ]2;corikomiyama@coris-mbp:~/desktop/coffee-mesh-project/ch02]1;..-project/ch02]7;file://coris-mbp.dhcp.asu.edu/Users/corikomiyama/desktop/coffee-mesh-project/ch02\[0m[27m[24m[J(ch02) [01;32m➜  [36mch02[00m [01;34mgit:([31mmain[34m) [33m✗[00m [K[?1h=[?2004hcclear[?1l>[?2004l
]2;clear]1;clear[H[2J[1m[7m%[27m[1m[0m                                                                                                                                                            ]2;corikomiyama@coris-mbp:~/desktop/coffee-mesh-project/ch02]1;..-project/ch02]7;file://coris-mbp.dhcp.asu.edu/Users/corikomiyama/desktop/coffee-mesh-project/ch02\[0m[27m[24m[J(ch02) [01;32m➜  [36mch02[00m [01;34mgit:([31mmain[34m) [33m✗[00m [K[?1h=[?2004h[0m[27m[24m[J(ch02) [01;32m➜  [36mch02[00m [01;34mgit:([31mmain[34m) [33m✗[00m [0m[27m[24m[J(ch02) [01;32m➜  [36mch02[00m [01;34mgit:([31mmain[34m) [33m✗[00m uuvicorn orders.app:app --reload[?1l>[?2004l
]2;uvicorn orders.app:app --reload]1;uvicorn[32mINFO[0m:     Will watch for changes in these directories: ['/Users/corikomiyama/Desktop/coffee-mesh-project/ch02']
[32mINFO[0m:     Uvicorn running on [1mhttp://127.0.0.1:8000[0m (Press CTRL+C to quit)
[32mINFO[0m:     Started reloader process [[36m[1m5341[0m] using [36m[1mStatReload[0m
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 24, in <module>
    class CreateOrderSchema(BaseModel):
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 25, in CreateOrderSchema
    order: conlist(OrderItemSchema, min_items=1)    # We use pydantic's conlist type to define a list with at least one element
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: conlist() got an unexpected keyword argument 'min_items'
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
Process SpawnProcess-2:
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/_subprocess.py", line 78, in subprocess_started
    target(sockets=sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/asyncio/base_events.py", line 685, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 69, in serve
    await self._serve(sockets)
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/server.py", line 76, in _serve
    config.load()
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/config.py", line 433, in load
    self.loaded_app = import_from_string(self.app)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/.local/share/virtualenvs/ch02-5ME4VFfE/lib/python3.12/site-packages/uvicorn/importer.py", line 19, in import_from_string
    module = importlib.import_module(module_str)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/app.py", line 5, in <module>
    from orders.api import api
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/api.py", line 8, in <module>
    from orders.api.schemas import CreateOrderSchema # We import the pydantic models so we can use them for validation
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 27, in <module>
    class GetOrderSchema(CreateOrderSchema):
  File "/Users/corikomiyama/Desktop/coffee-mesh-project/ch02/orders/api/schemas.py", line 29, in GetOrderSchema
    created: datetime
             ^^^^^^^^
NameError: name 'datetime' is not defined. Did you forget to import 'datetime'
[33mWARNING[0m:  StatReload detected changes in 'orders/api/schemas.py'. Reloading...
[32mINFO[0m:     Started server process [[36m5382[0m]
[32mINFO[0m:     Waiting for application startup.
[32mINFO[0m:     Application startup complete.
[32mINFO[0m:     127.0.0.1:50607 - "[1mGET /docs HTTP/1.1[0m" [32m200 OK[0m
[32mINFO[0m:     127.0.0.1:50607 - "[1mGET /openapi.json HTTP/1.1[0m" [32m200 OK[0m
